xdsl/parser/base_parser.py

```diff
@@ -350,8 +350,11 @@ def parse_integer(
             "Expected integer literal" + context_msg,
         )
 
-    def parse_optional_number(self) -> int | float | None:
-        """Parse a (possibly negative) integer or float literal, if present."""
+    def parse_optional_number(self, allow_boolean: bool = False) -> int | float | None:
```

Comment:
nit:
```suggestion
    def parse_optional_number(self, *, allow_boolean: bool = False) -> int | float | None:
```
To make the `allow_boolean` kw mandatory